fenixedu-ist-integration/src/main/java/pt/ist/fenixedu/integration/task/exportData/humanResources/ExportEmployeeInfo.java

```diff
@@ -54,11 +53,10 @@ public void runTask() throws Exception {
             final Person person = user.getPerson();
             if (person != null && user.getProfile() != null) {
                 final Employee employee = person.getEmployee();
-                final Researcher researcher = person.getResearcher();
-                if (employee != null && isActiveContractedTeacher(person)) {
+                if (employee != null && getCurrentContractedContractSituation(person, CategoryType.TEACHER) != null) {
                     registerContractSituation(result, user, person, employee, CategoryType.TEACHER);
                 }
-                if (researcher != null && researcher.isActiveContractedResearcher()) {
+                if (employee != null && getCurrentContractedContractSituation(person, CategoryType.RESEARCHER) != null) {
```

Comment:
Not exactly... the difference is that not all cases have a researcher object.
From what I could gather the researcher object is no longer created when stuff is imported from the human resources database, so now we have to only use the employee object.